<img src='https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQ-VfNtOyJbsaxu43Kztf_cv1mgBG6ZIQZEVw&usqp=CAU'>

# Procesamiento de Lenguage Natural

## Taller #2: Adquisición de textos
`Fecha de entrega: Marzo 4, 2021. (Antes del inicio de la próxima clase).`

# Punto 1:

- `[18 pts]` Descomprimir el archivo `.zip` de `python_books`
- `[22 pts]` Leer cada uno de sus archivos
- `[10 pts]` Responder: ¿Cuál archivo tiene el mayor número de palabras?

In [18]:
import os
from zipfile import ZipFile

In [33]:
direccion = "G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/python_books.zip"

In [35]:
with ZipFile (direccion) as archivo:
    archivo.extractall("G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2")

In [39]:
import shutil
shutil.rmtree("G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/__MACOSX")

In [40]:
for archivo in os.walk("G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/python_books"):
    print(archivo)
    
for raiz, dirs, archivos in os.walk("G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/python_books"):
    # print(raiz, dirs, archivos)	
    for a in archivos: 
        print(a)

('G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/python_books', [], ['Python - Finance.pdf', 'Python  Data Science Cookbook.pdf', 'Python - AWS.pdf'])
Python - Finance.pdf
Python  Data Science Cookbook.pdf
Python - AWS.pdf


In [77]:
import fitz

In [80]:
documento = fitz.open("G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/python_books/Python - Finance.pdf")

In [81]:
documento1 = fitz.open("G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/python_books/Python - AWS.pdf")

In [85]:
documento2 = fitz.open("G:/Mi unidad/NANI/_ESTUDIOS/_Especialización/Semestre 2/NLP/Clase 2/Taller 2/python_books/Python  Data Science Cookbook.pdf")

In [113]:
pagina = documento.loadPage(0)
texto = pagina.getText("text")
palabras = texto.split()
Num_palabras = len(palabras)
print ("El documento Python - Finance contiene: " + str(Num_palabras) + " palabras")

El documento Python - Finance contiene: 226 palabras


In [120]:
pagina1 = documento1.loadPage(0)
texto1 = pagina1.getText("text")
palabras1 = texto1.split()
Num_palabras1 = len(palabras1)
print ("El documento Python - AWS contiene: " + str(Num_palabras1) + " palabras")

El documento Python - AWS contiene: 254 palabras


In [123]:
pagina2 = documento2.loadPage(0)
texto2 = pagina2.getText("text")
palabras2 = texto2.split()
Num_palabras2 = len(palabras2)
print ("El documento Python  Data Science Cookbook contiene: " + str(Num_palabras2) + " palabras")

El documento Python  Data Science Cookbook contiene: 535 palabras


# El archivo que tiene mayor número de palabras es el "Python  Data Science Cookbook" que tiene 535 palabras